### PR TITLE
feat(compiler): expand table expression to its columns

### PR DESCRIPTION
**Pull request by rtpsw**
See https://github.com/ibis-project/ibis-substrait/issues/184

### CODE REVIEW

**Comment by rtpsw** [comment]
Note that one alternative is to make the table expansion a choice controlled by a parameter of `ibis.mutate` placed in a member of `ops.Selection`.

File: ibis_substrait/compiler/translate.py [hunk 1]
```diff
@@ -647,6 +647,11 @@ def selection(
         )
 
     # projection
+    selections = [
+        x
+        for s in op.selections
+        for x in (s.get_columns(s.columns) if isinstance(s, ir.TableExpr) else [s])
```
Comment (cpcloud) [inline ibis_substrait/compiler/translate.py:653]:
```suggestion
        col
        for sel in op.selections
        for col in (sel.get_columns(sel.columns) if isinstance(sel, ir.TableExpr) else [sel])
```

**Review by cpcloud — CHANGES_REQUESTED** [review]
@rtpsw Can you add a test for this change?

**Comment by rtpsw** [comment]
> @rtpsw Can you add a test for this change?

Will do.

**Comment by cpcloud** [comment]
@rtpsw This looks reasonable to me.

**Review by cpcloud — APPROVED** [review]
LGTM, just needs a squash then it's good to go.

**Pull request merged**